src/anyio/_core/_asyncio_selector_thread.py

```diff
@@ -114,7 +114,12 @@ def run(self) -> None:
         while not self._closed:
             for key, events in self._selector.select():
                 if key.fileobj is self._receive:
-                    self._receive.recv(4096)
+                    while True:
+                        try:
+                            self._receive.recv(4096)
```

Comment:
You need to make the socket non blocking

Reply:
But I did! Lines 22 and 23.